mne/io/eeglab/eeglab.py

```diff
@@ -0,0 +1,375 @@
+        :func:`mne.channels.read_montage` for more information.
+    eog : list or tuple
+        Names of channels or list of indices that should be designated
+        EOG channels. If None (default), the channel names beginning with
+        ``EOG`` are used.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    raw : Instance of RawEEGLAB
+        A Raw object containing EEGLAB .set data.
+
+    Notes
+    -----
+    .. versionadded:: 0.11.0
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    @verbose
+    def __init__(self, input_fname, montage, eog=None, preload=False,
+                 verbose=None):
+        """Read EEGLAB .set file.
+        """
+        from scipy import io
+        basedir = op.dirname(input_fname)
+        eeg = io.loadmat(input_fname, struct_as_record=False,
+                         squeeze_me=True)['EEG']
+        if eeg.trials != 1:
+            raise TypeError('The number of trials is %d. It must be 1 for raw'
+                            ' files. Please use `mne.io.read_epochs_eeglab` if'
+                            'the .set file contains epochs.' % eeg.trials)
+
+        last_samps = [eeg.pnts - 1]
+        info = _get_info(eeg, montage, eog)
+        # read the data
+        if isinstance(eeg.data, string_types):
+            data_fname = op.join(basedir, eeg.data)
+            logger.info('Reading %s' % data_fname)
+
+            super(RawEEGLAB, self).__init__(
+                info, preload, filenames=[data_fname], last_samps=last_samps,
+                orig_format='double', verbose=verbose)
+        else:
+            if not preload:
+                warnings.warn('Data will be preloaded. preload=False is not '
+                              'supported when the data is stored in the .set '
+                              'file')
+            data = eeg.data.reshape(eeg.nbchan, -1, order='F')
+            data = data.astype(np.double)
+            super(RawEEGLAB, self).__init__(
+                info, data, filenames=[input_fname], last_samps=last_samps,
+                orig_format='double', verbose=verbose)
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a chunk of raw data"""
+        n_bytes = 4
+        nchan = self.info['nchan']
+        data_offset = self.info['nchan'] * start * n_bytes
+        data_left = (stop - start) * nchan
+        # Read up to 100 MB of data at a time.
+        n_blocks = 100000000 // n_bytes
```

Comment:
If you actually want it to be 100 MB blocks, this needs to be `25000000 // n_bytes` because there are 4 bytes per sample. Right now it's 400 MB chunks I think